src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -871,14 +871,21 @@ internal void ResolveConflict([NotNull]string docID)
                         WriteLog.To.Database.I(Tag, "Resolving doc '{0}' (mine={1} and theirs={2})",
                             new SecureLogString(docID, LogMessageSensitivity.PotentiallyInsecure), localDoc.RevID,
                             remoteDoc.RevID);
-                        var resolvedDoc = ResolveConflict(localDoc, remoteDoc);
+                        conflictResolver = conflictResolver ?? new DefaultConflictResolver();
+                        var conflict = new Conflict(localDoc.IsDeleted ? null : localDoc, remoteDoc.IsDeleted ? null : remoteDoc);
+                        var resolvedDoc = conflictResolver.Resolve(conflict);
+                        if (resolvedDoc != null && resolvedDoc.Id != docID)
+                            throw new InvalidOperationException($"Resolved docID {resolvedDoc.Id} does not match with docID {docID}");
+                        if (resolvedDoc != null && resolvedDoc.Database != this)
+                            throw new InvalidOperationException($"Resolved document db {resolvedDoc.Database.Name} is different from expected db {this.Name}");
                         SaveResolvedDocument(resolvedDoc, localDoc, remoteDoc);
                     } finally {
                         localDoc?.Dispose();
                         remoteDoc?.Dispose();
                     }
-                } catch (Exception) {
+                } catch (Exception ex) {
                     success = false;
+                    WriteLog.To.Sync.W(Tag, $"Exception in conflict resolver: {ex.Message}.");
                     throw;
```

Comment:
Didn't the doc comments say this is supposed to be handled?  Do we have a test that throws an exception in the conflict handler?

Reply:
No tests for testing exception yet. Will add them with other advanced test cases in other PR.

Reply:
But without them there is no way of knowing if this is correct, which it seems like it is not.

Reply:
This log statement appears unneeded since it is logged again further up.  One of them should be removed (the other is in Replication.cs)